lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -824,7 +824,13 @@ sub investigate {
     my $investigation = $job->investigate;
     $self->respond_to(
         json => {json => $investigation},
-        html => {text => join("\n", map { "$_: $investigation->{$_}" } sort keys %$investigation)});
+        html => {
+            data => join(
+                "\n",
+                map {
+                    "<tr><td style=\"vertical-align: top;\"><b>$_:</b></td><td><pre>$investigation->{$_}</pre></td><tr>"
```

Comment:
This still requires proper escaping. Just use a template or use the JSON version and make the DOM tree client-side. I'd even go for the last alternative.

Reply:
What would be the benefit of the last alternative? Can you propose how it could be done?

Reply:
To me writing some JavaScript/DOM code seems just easier.

Reply:
can you propose the actual code?

Reply:
I'd just remove the HTML representation and use the JSON from JavaScript.

Reply:
@kraih Yes, that's what I've meant with "make the DOM tree client-side".

@okurz I could propose code but then I would basically taking over the ticket by implementing the whole thing myself.

Reply:
I don't mind if you implement it